### PR TITLE
fix: backport patch to sync exposed crypto

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/16812.

Backports https://boringssl-review.googlesource.com/c/boringssl/+/33984 to our patch set in order to correctly enumerate and instantiate ciphers. Also adds a test to ensure that `DES-EDE-CBC` actually works.

cc @nornagon 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Added a patch to fix incorrect enumeration and instantiation of Node.js ciphers in the Crypto module.

### CODE REVIEW

**Comment by nornagon** [comment]
FYI, that patch may change a little still before it gets merged upstream so we may need to update this.

File: spec/node-spec.js [hunk 1]
```diff
@@ -462,6 +462,14 @@ describe('node feature', () => {
     it('should be able to create an aes-256-cfb cipher', () => {
       require('crypto').createCipheriv('aes-256-cfb', '0123456789abcdef0123456789abcdef', '0123456789abcdef')
     })
+
+    it('should list des-ede-cbc in getCiphers', () => {
+      expect(require('crypto').getCiphers()).to.include('des-ede-cbc')
+    })
+
+    it('should be able to create an des-ede-cbc cipher', () => {
+      require('crypto').createCipheriv('des-ede-cbc', '0123456789abcdeff1e0d3c2b5a49786', 'fedcba9876543210')
```
Comment (nornagon) [inline spec/node-spec.js]: I think the IV needs to be a different length here. try 8 or 32?
```suggestion
      require('crypto').createCipheriv('des-ede-cbc', '0123456789abcdeff1e0d3c2b5a49786', 'fedcba98')
```

Comment (codebytere) [reply spec/node-spec.js]: hmm: i pulled that key and IV from Chromium's own tests [here](https://cs.chromium.org/chromium/src/third_party/boringssl/src/crypto/cipher_extra/test/cipher_tests.txt?q=des-ede-cbc&sq=package:chromium&dr=C&l=94)

Comment (nornagon) [reply spec/node-spec.js]: huh, that is odd then. it'd be worth tracking down what exactly is going on & why node things this is the wrong IV length!

Comment (codebytere) [reply spec/node-spec.js]: @ryzokuken any ideas?

Comment (nornagon) [reply spec/node-spec.js]: I'm not really sure what Chromium's tests are doing there, but I'm looking at [the definition of `EVP_des_ede_cbc`](https://github.com/google/boringssl/blob/15ba2d11a9873cb40b2a45bba18242fdb16ca4c6/crypto/fipsmodule/cipher/e_des.c#L183-L193) and it looks like the IV and keys should be half the length this test sets them to:

```cpp
DEFINE_METHOD_FUNCTION(EVP_CIPHER, EVP_des_ede_cbc) {
  // ...
  out->key_len = 16;
  out->iv_len = 8;
  // ...
}
```

Comment (nornagon) [reply spec/node-spec.js]: oh, you know what it is? Chrome's tests encode the bytes as hex, which are 2 chars per byte.

Comment (nornagon) [reply spec/node-spec.js]:
```suggestion
      require('crypto').createCipheriv('des-ede-cbc', Buffer.from('0123456789abcdeff1e0d3c2b5a49786', 'hex'), Buffer.from('fedcba9876543210', 'hex'))
```

**Comment by codebytere** [comment]
@nornagon @ryzokuken all set, ptal when you can 🙇‍♀️

**Comment by codebytere** [comment]
Flake on linux tests that is unrelated! Merging.

**Pull request merged**